packages/repository/test/unit/decorator/repository.ts

```diff
@@ -85,20 +85,21 @@ describe('repository decorator', () => {
     }
     ctx.bind('controllers.Controller2').toClass(Controller2);
 
-    const myController = await ctx.get<MyController>('controllers.Controller2');
+    const myController = await ctx.get<Controller2>('controllers.Controller2');
     expect(myController.noteRepo).to.be.not.null();
   });
 
   it('supports @repository(model, dataSource)', async () => {
     class Controller3 {
       constructor(
         @repository(Note, ds)
-        public noteRepo: Repository<Note>,
+        public noteRepo: EntityCrudRepository<Note, number>,
       ) {}
     }
     ctx.bind('controllers.Controller3').toClass(Controller3);
-    const myController = await ctx.get<MyController>('controllers.Controller3');
+    const myController = await ctx.get<Controller3>('controllers.Controller3');
     expect(myController.noteRepo).to.be.not.null();
+    await myController.noteRepo.find();
```

Comment:
Is there a better way to test for this? In #1182, the function hangs after it gets called. If there's a way to test that calling the function doesn't result in it being stuck, I'd be happy.